packages/web/src/components/userprofile/UserProfile.tsx

```diff
@@ -40,31 +48,46 @@ export const UserProfile: React.FC<UserProfileProps> = ({ user, setUser }: UserP
     setUser(newUser);
   };
 
-  return (
-    <Box border="1px" borderColor="gray.200" boxShadow="base" p={3}>
-      {user.discordId ? (
-        <>
-          <Text fontSize="xl"> Your discord account is linked with: {user.discordId} </Text>
-          <Button size="md" colorScheme="blue" onClick={unlinkDiscordId}>
-            Unlink Discord Account
-          </Button>
-          <Text>{errorMessage}</Text>
-        </>
-      ) : (
-        <>
-          <Text fontSize="xl">
-            {' '}
-            After linking you&apos;ll be able to join our community and talk with others from
-            Project X on a variety of topics.{' '}
-          </Text>
-          <Button as="a" size="sm" colorScheme="blue" href="/api/auth/discord/login">
-            Login with Discord
-          </Button>
-        </>
-      )}
-      <Text fontSize="6xl">{user.name}</Text>
-      <Text fontSize="xl">{user.pronouns}</Text>
-      <Text fontSize="xl">{user.schoolName}</Text>
-    </Box>
-  );
+  if (editToggle)
+    return (
+      <Box border="1px" borderColor="gray.200" boxShadow="base" p={3}>
+        <EditUserForm setEditToggle={setEditToggle} setUser={setUser} user={user} />
+      </Box>
+    );
+  else {
+    const editButton = isCurrentUser ? (
+      <Button colorScheme="blue" onClick={() => setEditToggle(true)}>
+        Edit
+      </Button>
+    ) : null;
+
+    return (
+      <Box border="1px" borderColor="gray.200" boxShadow="base" p={3}>
+        {user.discordId ? (
+          <>
+            <Text fontSize="xl"> Your discord account is linked with: {user.discordId} </Text>
+            <Button size="md" colorScheme="blue" onClick={unlinkDiscordId}>
+              Unlink Discord Account
+            </Button>
+            <Text>{errorMessage}</Text>
+          </>
+        ) : (
+          <>
+            <Text fontSize="xl">
+              {' '}
```

Comment:
Prettier thinks you meant to add this space but it was probably accidental since I don't see a reason it should be here
```suggestion
```